src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsBrowserError.java

```diff
@@ -0,0 +1,47 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.testng.annotations.Test;
+
+/**
+ * @ownership AdEng
+ */
+public class TestAdsBrowserError extends TemplateNoFirstLoad {
+
+  @Test(groups = "AdsScrollHandlerBrowserErrorMercury")
+  public void adsScrollHandlerBrowserErrorMercury() throws InterruptedException {
+    String testPage = urlBuilder.getUrlForPath("adtest", "Wikia_Ad_Testing");
+    testPage = urlBuilder.appendQueryStringToURL(testPage, "scrollhandler=1");
+    AdsBaseObject adsBaseObject = new AdsBaseObject(driver);
+    adsBaseObject.getUrl(testPage);
+
+    executeScript(addErrorListenerScript());
+    executeScript(scroll());
+
+    adsBaseObject.waitForJavaScriptTruthy("window.scrollY != 0");
+    Assertion.assertEquals(executeScript(getErrors()), "");
+  }
+
+  private String addErrorListenerScript() {
```

Comment:
Thanks @justnpT !

> Storing scripts as String in the code doesn't make it easier.

Completely agree with this statement.
Any suggestions how to store JS code as not a String type inside Java repo?
I can think only about using another Selenium bindings - JavaScript binding.

> Have you considered moving javascript related logic to some JavaScript handler class ?

This kind of test is unique if I may say. I would rather move JS logic to `JavascriptActions` class when we will have 2 cases where we can apply such logic.